rules/falco-incubating_rules.yaml

```
@@ -1252,3 +1252,15 @@
   output: Adding ssh keys to authorized_keys (file=%fd.name evt_type=%evt.type user=%user.name user_uid=%user.uid user_loginuid=%user.loginuid process=%proc.name proc_exepath=%proc.exepath parent=%proc.pname command=%proc.cmdline terminal=%proc.tty exe_flags=%evt.arg.flags)
   priority: WARNING
   tags: [maturity_incubating, host, filesystem, mitre_persistence, T1098.004]
+
+# Detection for possible use of CVE-2023-4911
+# Based on https://www.qualys.com/2023/10/03/cve-2023-4911/looney-tunables-local-privilege-escalation-glibc-ld-so.txt
+- rule: Program possibly trying to use CVE-2023-4911
```

```suggestion
- rule: Potential Local Privilege Escalation via Environment Variables Misuse
```

Suggestion: Let's start a more generic rule that can span beyond just one CVE as we are limited in the number of rules we can maintain and this is the reason why we are looking to push for new rules that are more generic and behavior based.

@loresuso is looking into a rule around `LD_PRELOAD` -> Lorenzo do you see potential to consolidate your work into this rule?

---

Makes sense, though I can't think of other environment variables leading to attacks off the top of my head... 
Are you wanting a list of environment variables that are known to be misused and this alert triggers if it detects one?

---

@loresuso mind posting a preview of the idea of the rule you have in mind around `LD_PRELOAD`?

Instead of a list of env variables it likely needs to be more granular and macros for each case, perhaps the sandbox rule `Decoding Payload in Container` is a good example. Currently it's just one use case but it's designed in a way that allows for easy expansion like `and (base64_decoding or other_decoding or other_macro)`. WDYT?

Again the reasoning is that we cannot for example go up to 500 or so rules, there is an upper bound and some consolidation may make sense. Also please note some older rules are on our radar for either replacement, deprecation or re-writing them as they don't align with what I just said.